src/graphql/resolvers.ts

```diff
@@ -28,6 +28,24 @@ export const resolvers = {
 
       return user;
     },
+    users: async (_: any, args: { numberOfUsers?: number }, context) => {
+      if (!context.id) {
+        throw new CustomError('Invalid token.', 401);
+      }
+
+      const users = await User.find({
+        order: {
+          name: 'ASC',
+        },
+        take: args.numberOfUsers || 5,
+      });
+
+      if (users.length === 0) {
+        throw new CustomError('Users not found', 404);
+      }
```

Comment:
Acho que aqui vc poderia retornar array vazio mesmo. É difícil ver sistemas que jogam erro de not-found quando não tem elementos. O not-found é mais usado pra quando vc busca por um id ou algum outro parâmetro e não acha.